woocommerce/woocommerce-ios
# Variations: Show screen before creating the first variation

closes #4390 

# Why

As part of the Variation changes, the design team has decided to include an intermediary screen before creating the first variation to explain to the user what is happening in a better fashion.

# How

- Use the `EmptyStateViewController` class to show the intermediate screen before creating the first variation.

- Update `EmptyStateViewController` so it receives a `config` during initialization so it can be used with the `show(:) and present(:)` UIKit methods. Before it would have crashed because the `configure(:)` method does not checks if the view is loaded before accessing its outlets.

# Demo

https://user-images.githubusercontent.com/562080/121747396-fbb4c780-cacc-11eb-8848-d4d76cf3dd83.mov

# Testing Steps

- Go to a variable product without attributes or variations
- Create your first variation.
- See that before creating the variation there is an intermediary screen telling you that you are ready to create your first variation!

Update release notes:
